homeassistant/components/utility_meter/__init__.py

```diff
@@ -150,7 +150,8 @@ async def async_reset_meters(service_call):
                     DOMAIN,
                     {meter: {CONF_METER: meter}},
                     config,
-                )
+                ),
+                eager_start=True,
```

Comment:
We shouldn't use this for `discovery.async_load_platform` until we've at minimum updated its docstring since it says it should use a task to avoid a deadlock. An eager task won't avoid a deadlock.

Reply:
I'm not sure why it would deadlock here as the eager task can still suspend if it needs to.

Reply:
Right, that makes sense. But it's still confusing to run this in an eager task if we know there's a deadlock coming. Or is the deadlock not always happening? I think we should clarify the docstring what is happening and how we need to take care to avoid the deadlock.

Reply:
The deadlock is going to be extremely rare.  I can't actually come up with a case where it could deadlock anymore.

Reply:
Adjusted the comment in https://github.com/home-assistant/core/pull/114771